platform/linux-generic/include/odp/api/plat/atomic_inlines.h

```diff
@@ -530,6 +534,140 @@ _ODP_INLINE int odp_atomic_cas_acq_rel_u32(odp_atomic_u32_t *atom,
 					   __ATOMIC_RELAXED);
 }
 
+#ifdef __ARM_FEATURE_ATOMICS
```

Comment:
ARM macros and assembly code belongs under linux-generic/arch/aarch64

Reply:
Hi Petri,
   Few questions:
  1.  Which file can I use under "~/ linux-generic/arch/aarch64/" directory to move arm specific assembly code ? Can I move the lockless macros into odp_atomic.h file (or) I have to create a new file under "~/ linux-generic/arch/aarch64/odp/api/abi" ? What are all the relevant make file changes ? 
  2.  I have used __ARM_FEATURE_ATOMICS predefined macro to differentiate lock Vs lockless variants of 128 bit atomics in "~/odp/api/plat/atomic_inlines.h" file. There is no GCC predefined macro available now to detect lockless 128 bit variant in ARM ISA. What is the alternate option to remove ARM specific predefined macro from atomic_inlines.h file ?

Reply:
> 1. Which file can I use under ~/ linux-generic/arch/aarch64/ directory to move arm specific assembly code ? Can I move the lockless macros into odp_atomic.h file (or) I have to create a new file under ~/ linux-generic/arch/aarch64/odp/api/abi" ? What are all the relevant make file changes ?

On a quick glance `platform/linux-generic/arch/aarch64/odp_atomic.h` looks like the right place. And for the generic implementation `platform/linux-generic/arch/default/odp_atomic.h`. The relevant Makefile is `platform/linux-generic/Makefile.am` and there you can check `odp_cpu.h` for example.

> 2. I have used __ARM_FEATURE_ATOMICS predefined macro to differentiate lock Vs lockless variants of 128 bit atomics in "~/odp/api/plat/atomic_inlines.h" file. There is no GCC predefined macro available now to detect lockless 128 bit variant in ARM ISA. What is the alternate option to remove ARM specific predefined macro from atomic_inlines.h file ?

To remove __ARM_FEATURE_ATOMICS define usage outside the arm arch files you could follow the same model as in include/odp/api/abi-default/cpu.h. E.g. use define _ODP_LOCK_FREE_128BIT_ATOMICS, which would in the first phase be defined only for arm64 in platform/linux-generic/arch/aarch64/odp/api/abi/atomic.h.

 If removing ARM specific code from `platform/linux-generic/include/odp/api/plat/atomic_inlines.h` is tricky, you could instead implement the new functions in `platform/linux-generic/odp_atomic.c`. The inlining optimization can be done in a later PR.